### ISSUE TITLE
Problem with external plugin's version upgrade

### ISSUE THREAD

**Issue by jusabatier** [georchestra/mapstore2-georchestra#494]
I don't know if I upgrade external plugins the right way : 

1. Open any context with the plugin enabled
2. Delete the plugin via the plugins tab
3. Upload and install the new version of the plugin via the UI
4. Re-enable the plugin in the context
5. Save the context

With this, I always have some problems : 

1. When reinstall the new plugin version and activate it in the context, it show the previous plugin's config, but when saving the default one is applied. I must delete and paste manually the old config to apply it correctly.

Is it the right way to deploy a new plugin version ?
Does other users have the same problems ?

I think it would be easier if we can directly replace the plugin by uploading an other version of it without having to delete it.

**Comment by Gaetanbrl** [comment]
> I must delete and paste manually the old config to apply it correctly.

I think some troubles exists about the plugin's config and plugin replacement (or weird behaviors if not identify as trouble).

Each time i need to replace a same plugin (e.g upgrade) without change the browser window, i need to copy / past the config to keep or force the config refresh. 

I dunno why but if i skip this copy / past step, the config seems not correctly apply or not apply with the correct "state".

I use "state" word because i suspect this is due to the react + state component flow... not really easy to manage.

**Comment by tdipisa** [comment]
@jusabatier @Gaetanbrl 
to install a new version of one extension it is enough to:

- open a context wizard containing that extension
- delete the extension that you need to update and install the new one

Doing that, the new extension version is automatically used also by other contexts that are using it. To be honest, we currently installed some new version of extensions we are working on without experiencing any issue with this procedure.

@offtherailz do you have some further feedback on this?

**Comment by Gaetanbrl** [comment]
> Doing that, the new extension version is automatically used also by other contexts that are using it

I agree, my feeback concern config section only. I need to copy / past again on each plugin replacement. Not major.

**Comment by tdipisa** [comment]
> > Doing that, the new extension version is automatically used also by other contexts that are using it
> 
> I agree, my feeback concern config section only. I need to copy / past again on each plugin replacement. Not major.

I understand. Feel free to open an issue on MS for an improvement. Thank you so much.

**Comment by jusabatier** [comment]
> uninstall the extension that you need to update and install the new version

Can you confirm me that we can exit context edition at this point and don't need to save it ?

**Comment by catmorales** [comment]
Yes for me it works without save the context.

**Comment by alexander-fedorenko** [comment]
I just did a quick check and confirm that uninstalling extension and installing new one is enough for update, context editing can be abandoned at this point. Custom configuration for extension will be unintacted in this case.

**Comment by tdipisa** [comment]
@catmorales @jusabatier I've just opened these two issues for future improvements. The one concerning the doc will be done soon.

https://github.com/geosolutions-it/MapStore2/issues/8453
https://github.com/geosolutions-it/MapStore2/issues/8454
